var/spack/repos/builtin/packages/bacio/package.py

```diff
@@ -39,3 +39,6 @@ def cmake_args(self):
         args = [self.define_from_variant("CMAKE_POSITION_INDEPENDENT_CODE", "pic")]
 
         return args
+
+    def patch(self):
+        filter_file(".*", "2.4.1", "VERSION", when="bacio@2.4.1")
```

Comment:
After looking at the definition of `filter_file`, I think you need to (there is no `when=` argument):
```suggestion
        if self.spec.satisifes("@2.4.1"):
            filter_file(".*", "2.4.1", "VERSION")
```